index.js

```diff
@@ -45,19 +48,21 @@ const oauthPlugin = fp(function (fastify, options, next) {
   const name = options.name
   const credentials = options.credentials
   const callbackUri = options.callbackUri
+  const callbackUriParams = options.callbackUriParams || {}
   const scope = options.scope
   const generateStateFunction = options.generateStateFunction || defaultGenerateStateFunction
   const checkStateFunction = options.checkStateFunction || defaultCheckStateFunction
   const startRedirectPath = options.startRedirectPath
 
   function startRedirectHandler (request, reply) {
     const state = generateStateFunction()
-
-    const authorizationUri = this[name].authorizationCode.authorizeURL({
+    const urlOptions = Object.assign(callbackUriParams, {
```

Comment:
This changes the original object. Could you change this line in order to avoid the initial object modification?

Reply:
Hi, I fixed it 👍 .